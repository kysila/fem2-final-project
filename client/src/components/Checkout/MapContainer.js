import {
  Box, Button, Grid, Typography,
} from '@material-ui/core';
import CloseIcon from '@material-ui/icons/Close';
import React from 'react';
import { useStyles } from './styles';
import Tungsten from '../../fonts/Tungsten-Book.woff';
import MuseoSans from '../../fonts/MuseoSans-500.woff';

const tungsten = {
  fontFamily: 'Tungsten Book',
  fontStyle: 'normal',
  src: `
    local('Tungsten Book'),
    url(${Tungsten}) format('woff')
  `,
};

const museo = {
  fontFamily: 'Museo Sans 500',
  fontStyle: 'normal',
  src: `
    local('Museo Sans 500'),
    url(${MuseoSans}) format('woff')
  `,
};

export function MapContainer(props) {
  const classes = useStyles();
  return (
    <Box
      className={classes.mapContainer}
    >
      <Grid
        container
        direction="column"
        alignItems="center"
        justify="flex-start"
        className={classes.mapBody}
      >
        <Typography style={{ fontSize: '36px', ...tungsten }}>
          Location Selector
        </Typography>
        <Typography style={{
          marginTop: '8px', marginBottom: '20px', fontSize: '16px', color: '#888888', textAlign: 'center', ...museo,
        }}
        >
          Drag the map to place locator on the needed place.
          <br />
          When ready - press “Done”. You can then edit address inside Address field.
        </Typography>
        {props.children}
        <Button
          style={{
            padding: '13px 42px', fontSize: '14px', marginTop: '20px', fontWeight: 'bold', ...museo,
          }}
          onClick={props.onClick}
        >
          Done
        </Button>
      </Grid>
      <Box className={classes.closeWrapper} onClick={props.onClick}>
        <CloseIcon className={classes.close} />
      </Box>
    </Box>
  );
}
