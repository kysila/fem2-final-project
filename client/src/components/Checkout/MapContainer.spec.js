/* eslint-disable */
import React from 'react';
import {shallow} from 'enzyme';
import {MapContainer} from './MapContainer';

describe('MapContainer component', () => {
  it('MapContainer render', () => {
    const rendered = shallow(
      <MapContainer onClick={() => null}/>,
    );

    expect(rendered).toMatchSnapshot();
  });
});
