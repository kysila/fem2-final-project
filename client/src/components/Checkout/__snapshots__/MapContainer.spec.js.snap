// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapContainer component MapContainer render 1`] = `
<Styled(MuiBox)
  className="makeStyles-mapContainer-19"
>
  <WithStyles(ForwardRef(Grid))
    alignItems="center"
    className="makeStyles-mapBody-20"
    container={true}
    direction="column"
    justify="flex-start"
  >
    <WithStyles(ForwardRef(Typography))
      style={
        Object {
          "fontFamily": "Tungsten Book",
          "fontSize": "36px",
          "fontStyle": "normal",
          "src": "
    local('Tungsten Book'),
    url(Tungsten-Book.woff) format('woff')
  ",
        }
      }
    >
      Location Selector
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Typography))
      style={
        Object {
          "color": "#888888",
          "fontFamily": "Museo Sans 500",
          "fontSize": "16px",
          "fontStyle": "normal",
          "marginBottom": "20px",
          "marginTop": "8px",
          "src": "
    local('Museo Sans 500'),
    url(MuseoSans-500.woff) format('woff')
  ",
          "textAlign": "center",
        }
      }
    >
      Drag the map to place locator on the needed place.
      <br />
      When ready - press “Done”. You can then edit address inside Address field.
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Button))
      onClick={[Function]}
      style={
        Object {
          "fontFamily": "Museo Sans 500",
          "fontSize": "14px",
          "fontStyle": "normal",
          "fontWeight": "bold",
          "marginTop": "20px",
          "padding": "13px 42px",
          "src": "
    local('Museo Sans 500'),
    url(MuseoSans-500.woff) format('woff')
  ",
        }
      }
    >
      Done
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(Grid))>
  <Styled(MuiBox)
    className="makeStyles-closeWrapper-23"
    onClick={[Function]}
  >
    <CloseIcon
      className="makeStyles-close-24"
    />
  </Styled(MuiBox)>
</Styled(MuiBox)>
`;
